libs/model/src/services/commonProtocol/aaWallet.ts

```diff
@@ -0,0 +1,75 @@
+import { createModularAccountAlchemyClient } from '@alchemy/aa-alchemy';
+import {
+  LocalAccountSigner,
+  SmartAccountSigner,
+  sepolia,
+} from '@alchemy/aa-core';
+import { AppError } from '@hicommonwealth/core';
+import Web3 from 'web3';
+
+const message =
+  'I approve commonwealth to create a smart wallet on behalf of this account';
+
+export const verifySignature = (
+  signerAddress: string,
+  signedMessage: string,
+) => {
+  const web3 = new Web3();
+  // Calculate the signer's address
+  const recoveredSignerAddress = web3.eth.accounts.recover(
+    message,
+    signedMessage,
+  );
+  if (signerAddress.toLowerCase() !== recoveredSignerAddress.toLowerCase()) {
```

Comment:
If #8020 is merged first, we need to update this to use the new address comparison utility.

Reply:
Will note to merge the address checker here before merging this PR

Reply:
@ianrowan on this and to fix CI issues/conflicts. Will approve promptly on Monday if completed.